integration-tests/src/test_loop/builder.rs

```diff
@@ -0,0 +1,294 @@
+use std::sync::{Arc, RwLock};
+
+use near_async::futures::FutureSpawner;
+use near_async::messaging::{noop, IntoMultiSender, IntoSender, LateBoundSender};
+use near_async::test_loop::sender::TestLoopSender;
+use near_async::test_loop::TestLoopV2;
+use near_async::time::{Clock, Duration};
+use near_chain::chunks_store::ReadOnlyChunksStore;
+use near_chain::runtime::NightshadeRuntime;
+use near_chain::state_snapshot_actor::{
+    get_delete_snapshot_callback, get_make_snapshot_callback, SnapshotCallbacks, StateSnapshotActor,
+};
+use near_chain::types::RuntimeAdapter;
+use near_chain::ChainGenesis;
+use near_chain_configs::{
+    ClientConfig, DumpConfig, ExternalStorageConfig, ExternalStorageLocation, Genesis,
+    StateSyncConfig, SyncConfig,
+};
+use near_chunks::shards_manager_actor::ShardsManagerActor;
+use near_client::client_actor::ClientActorInner;
+use near_client::sync_jobs_actor::SyncJobsActor;
+use near_client::test_utils::test_loop::test_loop_sync_actor_maker;
+use near_client::{Client, PartialWitnessActor, SyncAdapter};
+use near_epoch_manager::shard_tracker::{ShardTracker, TrackedConfig};
+use near_epoch_manager::EpochManager;
+use near_network::test_loop::TestLoopPeerManagerActor;
+use near_primitives::network::PeerId;
+use near_primitives::test_utils::create_test_signer;
+use near_primitives::types::AccountId;
+use near_store::config::StateSnapshotType;
+use near_store::genesis::initialize_genesis_state;
+use near_store::test_utils::create_test_store;
+use near_store::{StoreConfig, TrieConfig};
+use near_vm_runner::{ContractRuntimeCache, FilesystemContractRuntimeCache};
+use nearcore::state_sync::StateSyncDumper;
+use tempfile::TempDir;
+
+use super::env::{TestData, TestLoopEnv};
+
+pub struct TestLoopBuilder {
+    test_loop: TestLoopV2,
+    genesis: Option<Genesis>,
+    clients: Vec<AccountId>,
+}
+
+impl TestLoopBuilder {
```

Comment:
nit: could you please rearrange methods to have all public methods on top, this would make it a bit easier to discover its API